core/pkg/fixhandler/yamlhandler.go

```diff
@@ -0,0 +1,286 @@
+
+	var allDocuments = list.New()
+	reader, err := readStream(filePath)
+	if err != nil {
+		return yaml.Node{}
+	}
+
+	fileDocuments, err := readDocuments(reader, filePath, 0, decoder)
+	if err != nil {
+		return yaml.Node{}
+	}
+	allDocuments.PushBackList(fileDocuments)
+
+	if allDocuments.Len() == 0 {
+		candidateNode := &yqlib.CandidateNode{
+			Document:       0,
+			Filename:       "",
+			Node:           &yaml.Node{Kind: yaml.DocumentNode, Content: []*yaml.Node{{Tag: "!!null", Kind: yaml.ScalarNode}}},
+			FileIndex:      0,
+			LeadingContent: "",
+		}
+		allDocuments.PushBack(candidateNode)
+	}
+
+	allAtOnceEvaluator := yqlib.NewAllAtOnceEvaluator()
+
+	matches, _ := allAtOnceEvaluator.EvaluateCandidateNodes(yamlExpression, allDocuments)
+
+	return *matches.Front().Value.(*yqlib.CandidateNode).Node
+}
+
+func readStream(filename string) (io.Reader, error) {
+	var reader *bufio.Reader
+	if filename == "-" {
+		reader = bufio.NewReader(os.Stdin)
+	} else {
+		// ignore CWE-22 gosec issue - that's more targeted for http based apps that run in a public directory,
+		// and ensuring that it's not possible to give a path to a file outside thar directory.
+		file, err := os.Open(filename) // #nosec
+		if err != nil {
+			return nil, err
+		}
+		reader = bufio.NewReader(file)
+	}
+	return reader, nil
+
+}
+
+func readDocuments(reader io.Reader, filename string, fileIndex int, decoder yqlib.Decoder) (*list.List, error) {
+	err := decoder.Init(reader)
+	if err != nil {
+		return nil, err
+	}
+	inputList := list.New()
+	var currentIndex uint
+
+	for {
+		candidateNode, errorReading := decoder.Decode()
+
+		if errors.Is(errorReading, io.EOF) {
+			switch reader := reader.(type) {
+			case *os.File:
+				safelyCloseFile(reader)
+			}
+			return inputList, nil
+		} else if errorReading != nil {
+			return nil, fmt.Errorf("bad file '%v': %w", filename, errorReading)
+		}
+		candidateNode.Document = currentIndex
+		candidateNode.Filename = filename
+		candidateNode.FileIndex = fileIndex
+		candidateNode.EvaluateTogether = true
+
+		inputList.PushBack(candidateNode)
+
+		currentIndex = currentIndex + 1
+	}
+}
+
+func safelyCloseFile(file *os.File) {
+	err := file.Close()
+	if err != nil {
+		fmt.Println("Error Closing File")
+	}
+}
+
+func getLineAndContentToAdd(node *yaml.Node) *[]LineAndContentToAdd {
+	contentToAdd := make([]LineAndContentToAdd, 0)
+	getLineAndContentToAddHelper(0, node, &contentToAdd)
+	return &contentToAdd
+}
+
+func getLineAndContentToAddHelper(nodeIdx int, parentNode *yaml.Node, contentToAdd *[]LineAndContentToAdd) {
+	node := parentNode.Content[nodeIdx]
+	var content string
+	var err error
+	if node.Line == 0 && node.Column == 0 {
+		if parentNode.Kind == yaml.MappingNode {
+			if nodeIdx%2 != 0 {
+				return
+			}
+		}
+		content, err = enocodeIntoYaml(parentNode, nodeIdx)
+
+		if err != nil {
+			fmt.Println("Cannot Encode into YAML")
+		}
+
+		indentationSpacesBeforeContent := parentNode.Column - 1
```

Comment:
nitpick: variable name seems redundant
Since “indentation” already means “spaces before actual content”, we can just name this variable `indentationSpaces`.